atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,309 @@
+
+        # Get documents
+        docs = list(
+            mmdb.collection.find(
+                {"wf_meta.wf_uuid": wf_uuid}, ["heisenberg_model", "nn_cutoff"]
+            )
+        )
+
+        hmodels = [HeisenbergModel.from_dict(d["heisenberg_model"]) for d in docs]
+        cutoffs = [d["nn_cutoff"] for d in docs]
+
+        # Take <J> only if no static calcs
+        if self["average"]:
+            hmodel = hmodels[0].as_dict()
+        else:
+            # Check for J_ij convergence
+            converged_list = []
+            for cutoff, hmodel in zip(cutoffs, hmodels):
+                ex_params = hmodel.ex_params
+                converged = True
+
+                # J_ij exists
+                if len(ex_params) > 1:
+                    E0 = abs(ex_params["E0"])
+                    for k, v in ex_params.items():
+                        if k != "E0":
+                            if abs(v) > E0:  # |J_ij| > |E0| unphysical!
+                                converged = False
+                else:  # Only <J> was computed
+                    converged = False
+
+                if converged:
+                    converged_list.append(hmodel.as_dict())
+
+            # If multiple Heisenberg Models converged, take the maximal cutoff
+            if len(converged_list) > 0:
+                hmodel = converged_list[-1]  # Largest cutoff
+            else:  # No converged models
+                hmodel = None
+
+        # Update FW spec with converged hmodel or None
+        update_spec = {"converged_heisenberg_model": hmodel}
+
+        return FWAction(update_spec=update_spec)
+
+
+@explicit_serialize
+class VampireMC(FiretaskBase):
+    """
+    If there is a suitable Heisenberg model, run Monte Carlo to compute the
+    critical temperature.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+
+    Optional args:
+        mc_settings (dict): A configuration dict for monte carlo.
+            See pymatgen.command_line.VampireCaller for options.
+
+    TODO:
+        * Include HeisenbergModel convergence check.
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+    ]
+
+    optional_params = ["mc_settings"]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+        mc_settings = self.get("mc_settings", {})
+
+        # Get Heisenberg models from db
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["exchange"]
+
+        # Get documents
+        docs = list(
+            mmdb.collection.find(
+                {"wf_meta.wf_uuid": wf_uuid}, ["heisenberg_model", "nn_cutoff"]
+            )
+        )
+
+        hmodels = [HeisenbergModel.from_dict(d["heisenberg_model"]) for d in docs]
+        hmodel = hmodels[0]  # Take the model with smallest NN cutoff
```

Comment:
Is this guaranteed to be the model with the smallest cutoff? Should you sort the list on cutoff size first?